### PR TITLE
fix(mock-storage): fixed missing exceptions and exceptions format

### LINKED ISSUE #200: throw error in vmJs in del if key does not exist

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**
Aim's to resolve #200 and #184.

### CODE REVIEW

File: vm-mock/vm.js [hunk 1]
```diff
@@ -243,18 +241,20 @@ export default function createMockedABI(
...
       assembly_script_get_data(kPtr) {
-        let v = '';
         const k = ptrToUint8ArrayString(kPtr);
 
         if (ledger.has(contractAddress)) {
           const addressStorage = ledger.get(contractAddress).storage;
 
           if (addressStorage.has(k)) {
-            v = addressStorage.get(k);
+            return addressStorage.get(k);
```
Comment (Ben-Rey) [inline vm-mock/vm.js:250]: Here you don't use newArrayBuffer. It doesn't seem to affect the tests. Are you sure we don't need it ?

File: vm-mock/vm.js [hunk 2]
```diff
@@ -288,13 +288,17 @@ export default function createMockedABI(
         let v = '';
         const a = ptrToString(aPtr);
         const k = ptrToUint8ArrayString(kPtr);
+
         if (ledger.has(a)) {
           const addressStorage = ledger.get(a).storage;
           if (addressStorage.has(k)) {
-            v = addressStorage.get(k);
+            return addressStorage.get(k);
```
Comment (Ben-Rey) [inline vm-mock/vm.js:295]: same here

File: vm-mock/vm.js [hunk 3]
```diff
@@ -193,7 +193,6 @@ export default function createMockedABI(
    * Check if the address prefix is valid
```
Comment (Ben-Rey) [inline vm-mock/vm.js:193]: Just to trigger the review

**Review by Ben-Rey — CHANGES_REQUESTED** [review]

**Review by Ben-Rey — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]

**Comment by gregLibert** [comment]
@0xSwapFeeder can you fix the conflict please and reassign this PR to me ?

**Review by gregLibert — CHANGES_REQUESTED** [review]
Fix conflict

**Comment by 0xSwapFeeder** [comment]
> Fix conflict

Rebased branch on main :)

**Review by gregLibert — APPROVED** [review]

**Pull request merged**